Source/WebKit/WebProcess/Plugins/PDF/PDFIncrementalLoader.mm

```diff
@@ -649,19 +656,28 @@ static void dataProviderReleaseInfoCallback(void* info)
         return 0;
     }
 
-    WTF::Semaphore dataSemaphore { 0 };
+    Ref dataSemaphore = SemaphoreWrapper::create();
+    {
+        Locker locker { m_wasPDFThreadTerminationRequestedLock };
+        if (m_wasPDFThreadTerminationRequested)
+            return 0;
+
+        m_dataSemaphores.add(dataSemaphore.get());
+    }
+
     size_t bytesProvided = 0;
 
-    RunLoop::main().dispatch([protectedLoader = Ref { *this }, position, count, buffer, &dataSemaphore, &bytesProvided] {
-        protectedLoader->getResourceBytesAtPosition(count, position, [count, buffer, &dataSemaphore, &bytesProvided](const uint8_t* bytes, size_t bytesCount) {
+    RunLoop::main().dispatch([protectedLoader = Ref { *this }, dataSemaphore, position, count, buffer, &bytesProvided] {
+        protectedLoader->getResourceBytesAtPosition(count, position, [count, buffer, dataSemaphore, &bytesProvided](const uint8_t* bytes, size_t bytesCount) {
             RELEASE_ASSERT(bytesCount <= count);
             memcpy(buffer, bytes, bytesCount);
             bytesProvided = bytesCount;
```

Comment:
I "think" we should return early for protecting the case when it is already dispatched (after loader is destructed), or i will met crash due to BAD_ACCESS on this `bytesProvided` based on my local tests.

Reply:
That's a good point. I'll address.